src/components/Signup.js

```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { useInput } from '../hooks';
+import { authSignup } from '../actions';
+import SignupForm from './SignupForm';
+
+const Signup = ({ authSignup }) => {
+  const [email, setEmail, updateEmail] = useInput();
+  const [username, setUsername, updateUsername] = useInput();
+  const [password, setPassword, updatePassword] = useInput();
+  const [firstName, setFirstName, updateFirstName] = useInput();
+  const [lastName, setLastName, updateLastName] = useInput();
+
```

Comment:
These here, the `firstname` and `lastname` variables need to be camel-cased wherever they're in use.